src/java-opentimelineio/src/main/java/io/opentimeline/opentimelineio/AnyVector.java

```diff
@@ -0,0 +1,232 @@
+package io.opentimeline.opentimelineio;
+
+import io.opentimeline.OTIONative;
+import io.opentimeline.OTIOObject;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.TreeSet;
+import java.util.function.Consumer;
+
+public class AnyVector extends OTIOObject implements Collection<Any> {
```

Comment:
Collection is very nice kudos for implementing all of this boiler plate.
Alternatively might have been a bit less work to just implement Iterable and add a custom .size() method, but I am sure users of the library would find the rest very useful.